### PR TITLE
Fix bug of CAE new request disposing

### PR DESCRIPTION
**Pull request by msJinLei**
Request are used after disposed in AuthenticationHelper and so the previous request cannot be disposed.
The new request should be disposed finally.

- Normal case: smoke test coverred https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1385063&view=results
- 401 case: manual tests covered

## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Review by isra-fel — DISMISSED** [review]
LGTM. Have you tested CAE scenario?

File: src/Accounts/Accounts/CommonModule/ContextAdapter.cs [hunk 1]
```diff
@@ -195,30 +195,32 @@ public object GetParameterValue(string resourceId, string moduleName, Invocation
...
-                        if (!string.IsNullOrEmpty(claimsChallenge))
+                        try
                         {
-                            await processor.OnClaimsChallenageAsync(newRequest, claimsChallenge, cancelToken).ConfigureAwait(false);
-                            response = await next(newRequest, cancelToken, cancelAction, signal);
+                            var claimsChallenge = ClaimsChallengeUtilities.GetClaimsChallenge(response);
+                            if (!string.IsNullOrEmpty(claimsChallenge))
+                            {
+                                await processor.OnClaimsChallenageAsync(newRequest, claimsChallenge, cancelToken).ConfigureAwait(false);
+                                response = await next(newRequest, cancelToken, cancelAction, signal);
```
Comment (dolauli) [inline src/Accounts/Accounts/CommonModule/ContextAdapter.cs:213]: I double check the code and think we need to dispose the old response and request here.

**Review by isra-fel — DISMISSED** [review]
LGTM

File: src/Accounts/Accounts/CommonModule/ContextAdapter.cs [hunk 2]
```diff
@@ -195,30 +195,35 @@ public object GetParameterValue(string resourceId, string moduleName, Invocation
         internal async Task<HttpResponseMessage> AuthenticationHelper(IAzureContext context, string endpointResourceIdKey, string endpointSuffixKey, HttpRequestMessage request, CancellationToken cancelToken, Action cancelAction, SignalDelegate signal, NextDelegate next, TokenAudienceConverterDelegate tokenAudienceConverter = null)
         {
             IAccessToken accessToken = await AuthorizeRequest(context, request, cancelToken, endpointResourceIdKey, endpointSuffixKey, tokenAudienceConverter);
-            var newRequest = await request.CloneWithContentAndDispose(request.RequestUri, request.Method);
-            var response = await next(request, cancelToken, cancelAction, signal);
-
-            if (response.MatchClaimsChallengePattern())
+            using (var newRequest = await request.CloneWithContent(request.RequestUri, request.Method))
```
Comment (VeryEarly) [inline src/Accounts/Accounts/CommonModule/ContextAdapter.cs:198]:
```suggestion
            using (var newRequest = await request.CloneWithContent(request.RequestUri, request.Method))
            using (var response = await next(request, cancelToken, cancelAction, signal))
```

Comment (msJinLei) [reply src/Accounts/Accounts/CommonModule/ContextAdapter.cs:198]: Refer to the implementation of pipeline, the original request and response will be disposed finally.

**Review by VeryEarly — APPROVED** [review]

**Pull request merged**